pkg/admission/clusterworkspacetypeexists/admission.go

```diff
@@ -94,6 +91,35 @@ func (o *clusterWorkspaceTypeExists) Admit(ctx context.Context, a admission.Attr
 		return fmt.Errorf("failed to convert unstructured to ClusterWorkspace: %w", err)
 	}
 
+	if !o.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	clusterName, err := genericapirequest.ClusterNameFrom(ctx)
+	if err != nil {
+		return apierrors.NewInternalError(err)
+	}
+
+	cwt, err := o.typeLister.Get(clusters.ToClusterAwareKey(clusterName, strings.ToLower(cw.Spec.Type)))
+	if err != nil && apierrors.IsNotFound(err) {
+		if cw.Spec.Type == "Universal" {
+			return nil // Universal is always valid
```

Comment:
you better want to fall through. Otherwise the control flow gets tricky.

Reply:
If a cw is valid without the cwt for type Universal, that means there is no cwt defining additional labels to apply, which seems ok to me?

Reply:
Universal is the only type that does not have to exist to pass admission. Your code is fine. Was just complaining about the execution flow with this deeply nested return (which I introduced, didn't I?).

Reply:
Yes, all you 😄